strace assumes that if PT_GETSIGINFO is defined, you're building for ia64 as
this was the only architecture that exported PT_GETSIGINFO via glibc userspace
headers.  with newer glibc's, support for PT_GETSIGINFO has been added to all
linux architectures so we can no longer assume PT_GETSIGINFO == ia64.

http://bugs.gentoo.org/149945

--- strace.c	11 Oct 2006 22:55:25 -0000	1.71
+++ strace.c	11 Oct 2006 23:11:44 -0000	1.72
@@ -2261,7 +2261,7 @@ Process %d attached (waiting for parent)
 			if (!cflag
 			    && (qual_flags[WSTOPSIG(status)] & QUAL_SIGNAL)) {
 				unsigned long addr = 0, pc = 0;
-#ifdef PT_GETSIGINFO
+#if defined(PT_CR_IPSR) && defined(PT_CR_IIP) && defined(PT_GETSIGINFO)
 #				define PSR_RI	41
 				struct siginfo si;
 				unsigned long psr;
